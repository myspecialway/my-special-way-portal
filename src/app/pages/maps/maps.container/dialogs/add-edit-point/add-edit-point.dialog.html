<div class="container add-update-point-modal">
  <h3 mat-dialog-title>{{dialogTitle}}</h3>

<mat-divider></mat-divider>
 <mat-dialog-content>
  <form class="mat-dialog-content" [formGroup]="form" #formControl="ngForm">
    <div class="point-details">
      <mat-form-field color="primary">
        <input matInput #input formControlName="location_id" class="form-control" placeholder="נקודה" [(ngModel)]="data.location_id" name="location_id">
        <mat-error *ngIf="formControl.invalid">{{getErrorMessage()}}</mat-error>
      </mat-form-field>
      <mat-form-field color="primary">
        <input matInput #input formControlName="name" class="form-control" placeholder="שם" [(ngModel)]="data.name" name="name">
        <mat-error *ngIf="formControl.invalid">{{getErrorMessage()}}</mat-error>
      </mat-form-field>
      <mat-form-field color="primary">
        <mat-select placeholder="נקודה" formControlName="type" (ngModelChange)="onTypeSelect($event)" name="type">
            <mat-option disabled>בחר</mat-option>
            <mat-option *ngFor="let pointType of pointTypes" [value]="pointType">{{pointType}}</mat-option>
          </mat-select>
        <mat-error *ngIf="formControl.invalid">{{getErrorMessage()}}</mat-error>
      </mat-form-field>
    </div>
    <mat-button-toggle-group formControlName="type" name="icon" class="icons-list-container" aria-label="Map Point Icon">
      <mat-button-toggle *ngFor="let icon of icons; let idx = index;" [value]="icon" [checked]="idx===0" class="icons-container">
        <img src="../../../../../assets/MSW Symbols/{{icon}}.png" alt="{{icon}}">
      </mat-button-toggle>
    </mat-button-toggle-group>
  </form>
 </mat-dialog-content>
  <mat-dialog-actions>
    <button class="mat-button"  (click)="close(CANCEL)">ביטול</button>
    <button class="mat-raised-button mat-primary" (click)="close()">
      {{data.isNewPoint ? 'הוסף' :
    'עדכן'}}
      </button>
  </mat-dialog-actions>
</div>
