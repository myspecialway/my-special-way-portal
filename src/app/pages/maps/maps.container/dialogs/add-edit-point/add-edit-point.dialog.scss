* {
  direction: rtl;
  text-align: right;
}

.add-update-point-modal {
  width: 620px;
  height: 550px;
}

mat-dialog-actions button {
  text-align: center;
}

.mat-divider {
  border-top: 3px solid #8b66a7;
  margin-bottom: 28px;
}

.mat-dialog-actions {
  display: flex;
  flex-direction: row;
  width: 100%;
  justify-content: flex-end;
  margin-bottom: -12px;
}

.icons-list-container {
  display: block;
  width: 572px;
  height: 240px;
  background-color: #fafafa;
  padding: 9px;
  overflow-y: auto;

  .icons-container {
    width: 80px;
    height: 80px;
    border-radius: 2px;
    background-color: #fafafa;
    text-align: center;
    vertical-align: middle;
    float: right;
    margin: 5px;
    &.mat-button-toggle-checked {
      border-radius: 2px;
      border: solid 2px #8249d1;
    }
    img {
      width: 70px;
      height: 70px;
    }
  }
}

.point-details {
  display: flex;
  align-items: center;
  justify-content: space-between;

  mat-form-field {
    margin-left: 19px;
  }
}
