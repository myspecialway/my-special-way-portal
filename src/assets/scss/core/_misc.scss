body {
  background-color: #fafafa;
  color: $black-color;
  font-weight: 300;
}

legend {
  border-bottom: 0;
}

.serif-font {
  font-family: $font-family-serif;
}

// Prevent highlight on mobile
* {
  -webkit-tap-highlight-color: rgba(255, 255, 255, 0);
  -webkit-tap-highlight-color: transparent;
  &:focus {
    outline: 0;
  }
}

a {
  color: $link-color;
  &:hover,
  &:focus {
    color: darken($link-color, 5%);
    text-decoration: none;
  }

  &.text-info {
    &:hover,
    &:focus {
      color: darken($brand-info, 5%);
    }
  }

  & .material-icons {
    vertical-align: middle;
  }
}

.form-check,
label {
  font-size: 14px;
  line-height: 1.42857;
  color: $checkboxes-text-color;
  font-weight: 400;
}

/*           Animations              */
.animation-transition-general {
  @include transition-all($general-transition-time, $transition-linear);
}

.animation-transition-slow {
  @include transition-all($slow-transition-time, $transition-linear);
}

.animation-transition-fast {
  @include transition-all($fast-transition-time, $transition-ease);
}

.caret,
.sidebar a {
  @include transition-all($fast-transition-time, $transition-ease-in);
}
#map {
  position: relative;
  z-index: 2;
  width: 100%;
  height: calc(100vh - 70px);
  margin-top: 70px;
}

@font-face {
  font-family: 'pass';
  font-style: normal;
  font-weight: 400;
  src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATsAA8AAAAAB2QAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcg9+z70dERUYAAAF0AAAAHAAAAB4AJwANT1MvMgAAAZAAAAA/AAAAYH7AkBhjbWFwAAAB0AAAAFkAAAFqZowMx2N2dCAAAAIsAAAABAAAAAQAIgKIZ2FzcAAAAjAAAAAIAAAACAAAABBnbHlmAAACOAAAALkAAAE0MwNYJ2hlYWQAAAL0AAAAMAAAADYPA2KgaGhlYQAAAyQAAAAeAAAAJAU+ATJobXR4AAADRAAAABwAAAAcCPoA6mxvY2EAAANgAAAAEAAAABAA5gFMbWF4cAAAA3AAAAAaAAAAIAAKAE9uYW1lAAADjAAAARYAAAIgB4hZ03Bvc3QAAASkAAAAPgAAAE5Ojr8ld2ViZgAABOQAAAAGAAAABuK7WtIAAAABAAAAANXulPUAAAAA1viLwQAAAADW+JM4eNpjYGRgYOABYjEgZmJgBEI2IGYB8xgAA+AANXjaY2BifMg4gYGVgYVBAwOeYEAFjMgcp8yiFAYHBl7VP8wx/94wpDDHMIoo2DP8B8kx2TLHACkFBkYA8/IL3QB42mNgYGBmgGAZBkYGEEgB8hjBfBYGDyDNx8DBwMTABmTxMigoKKmeV/3z/z9YJTKf8f/X/4/vP7pldosLag4SYATqhgkyMgEJJnQFECcMOGChndEAfOwRuAAAAAAiAogAAQAB//8AD3jaY2BiUGJgYDRiWsXAzMDOoLeRkUHfZhM7C8Nbo41srHdsNjEzAZkMG5lBwqwg4U3sbIx/bDYxgsSNBRUF1Y0FlZUYBd6dOcO06m+YElMa0DiGJIZUxjuM9xjkGRhU2djZlJXU1UDQ1MTcDASNjcTFQFBUBGjYEkkVMJCU4gcCKRTeHCk+fn4+KSllsJiUJEhMUgrMUQbZk8bgz/iA8SRR9qzAY087FjEYD2QPDDAzMFgyAwC39TCRAAAAeNpjYGRgYADid/fqneL5bb4yyLMwgMC1H90HIfRkCxDN+IBpFZDiYGAC8QBbSwuceNpjYGRgYI7594aBgcmOAQgYHzAwMqACdgBbWQN0AAABdgAiAAAAAAAAAAABFAAAAj4AYgI+AGYB9AAAAAAAKgAqACoAKgBeAJIAmnjaY2BkYGBgZ1BgYGIAAUYGBNADEQAFQQBaAAB42o2PwUrDQBCGvzVV9GAQDx485exBY1CU3PQgVgIFI9prlVqDwcZNC/oSPoKP4HNUfQLfxYN/NytCe5GwO9/88+/MBAh5I8C0VoAtnYYNa8oaXpAn9RxIP/XcIqLreZENnjwvyfPieVVdXj2H7DHxPJH/2/M7sVn3/MGyOfb8SWjOGv4K2DRdctpkmtqhos+D6ISh4kiUUXDj1Fr3Bc/Oc0vPqec6A8aUyu1cdTaPZvyXyqz6Fm5axC7bxHOv/r/dnbSRXCk7+mpVrOqVtFqdp3NKxaHUgeod9cm40rtrzfrt2OyQa8fppCO9tk7d1x0rpiQcuDuRkjjtkHt16ctbuf/radZY52/PnEcphXpZOcofiEZNcQAAeNpjYGIAg///GBgZsAF2BgZGJkZmBmaGdkYWRla29JzKggxD9tK8TAMDAxc2D0MLU2NjENfI1M0ZACUXCrsAAAABWtLiugAA)
    format('woff');
}

input[type='password'] {
  font-family: 'pass', 'Roboto', Helvetica, Arial, sans-serif;
  font-size: 18px;
  &::-webkit-input-placeholder {
    transform: scale(0.77);
    transform-origin: 0 50%;
  }
  &::-moz-placeholder {
    font-size: 14px;
    opacity: 1;
  }
  &:-ms-input-placeholder {
    font-size: 14px;
    font-family: 'Roboto', Helvetica, Arial, sans-serif;
  }
}
