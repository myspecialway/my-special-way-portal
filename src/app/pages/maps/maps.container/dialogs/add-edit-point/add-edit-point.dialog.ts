import { MAP_POINT_TYPES } from './../../../maps-constants';
import { Location } from './../../../../../models/location.model';
import { Component, Inject, OnInit } from '@angular/core';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
import { FormControl, Validators, FormGroup, FormBuilder } from '@angular/forms';

@Component({
  selector: 'app-add-edit-point.dialog',
  templateUrl: './add-edit-point.dialog.html',
  styleUrls: ['./add-edit-point.dialog.scss'],
})
export class AddEditPointDialogComponent implements OnInit {
  dialogTitle = 'נקודת ניווט';
  CANCEL = true;
  pointTypes = MAP_POINT_TYPES;
  form: FormGroup;
  formControl = new FormControl('', [Validators.required]);
  constructor(
    private formBuilder: FormBuilder,
    public dialogRef: MatDialogRef<AddEditPointDialogComponent>,
    @Inject(MAT_DIALOG_DATA) public data: Location,
  ) {}
  public icons;
  public allicons = ['00026', '00036', '00040', '00047', '00067', '00135', '00319', '00336', '00468', '00524', '00545'];

  ngOnInit(): void {
    this.form = this.formBuilder.group({
      location_id: '',
      name: '',
      type: '',
      icon: '',
    });
    this.icons = this.allicons;
  }

  getErrorMessage() {
    return this.formControl.hasError('required') ? 'Required field' : '';
  }

  close(cancel = false): void {
    const response = cancel ? undefined : this.form.value;
    this.dialogRef.close(response);
  }

  onIconClick(selectedIcon: string): void {
    this.form.value.icon = selectedIcon;
  }

  onTypeSelect(ev) {}
}
