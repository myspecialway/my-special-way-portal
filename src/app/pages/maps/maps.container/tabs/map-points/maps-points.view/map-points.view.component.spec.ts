import { AddEditPointDialogComponent } from './../../../dialogs/add-edit-point/add-edit-point.dialog';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { mockedLocations } from './../../../../../../../mocks/assets/locations.mock';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MapPointsViewComponent } from './map-points.view.component';
import {
  MatListModule,
  MatSelectModule,
  MatSlideToggleModule,
  MatTableModule,
  MatInputModule,
  MatButtonModule,
  MatButtonToggleModule,
  MatGridListModule,
  MatDialogModule,
} from '@angular/material';

describe('MapPointsViewComponent', () => {
  let fixture: ComponentFixture<MapPointsViewComponent>;
  let component: MapPointsViewComponent;
  const beforeEachAsync = async () => {
    TestBed.configureTestingModule({
      imports: [
        MatButtonToggleModule,
        MatButtonModule,
        MatInputModule,
        MatSelectModule,
        MatDialogModule,
        MatTableModule,
        MatGridListModule,
        MatSlideToggleModule,
        MatListModule,
        BrowserAnimationsModule,
      ],
      // providers: [MatDialog],
      declarations: [MapPointsViewComponent, AddEditPointDialogComponent],
    }).compileComponents();
  };
  beforeEach(beforeEachAsync);

  beforeEach(() => {
    fixture = TestBed.createComponent(MapPointsViewComponent);
    component = fixture.componentInstance;
    component.locations = mockedLocations;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should update Location', () => {
    component.update.emit = jest.fn();

    const testLocation = {
      _id: '3',
      name: 'כיתת קג׳בו',
      disabled: false,
      location_id: '3',
      position: {
        latitude: 31.986419691740092,
        longitude: 34.91078563034535,
        floor: 1,
      },
    };

    component.onEdit(testLocation);
    expect(component.update.emit).toHaveBeenLastCalledWith({
      _id: '3',
      name: 'כיתת קג׳בו',
      disabled: false,
      location_id: '3',
      position: {
        latitude: 31.986419691740092,
        longitude: 34.91078563034535,
        floor: 1,
      },
    });
    // component.onLocationNameChanged(testLocation, 'test name');
    // expect(component.update.emit).toHaveBeenLastCalledWith({ _id: testLocation._id, name: 'test name' });
    // component.onLocationIdChanged(testLocation, '4');
    // expect(component.update.emit).toHaveBeenLastCalledWith({ _id: testLocation._id, location_id: '4' });
  });
});
