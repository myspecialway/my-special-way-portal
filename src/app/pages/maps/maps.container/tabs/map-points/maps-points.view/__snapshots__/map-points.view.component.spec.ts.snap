// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDetailsComponent with class info should match snapshot 1`] = `
<app-map-points-view
  delete={[Function EventEmitter]}
  dialog={[Function MatDialog]}
  displayedColumns={[Function Array]}
  locations={[Function Array]}
  update={[Function EventEmitter]}
>
  <div
    class="msw-table-container"
  >
    <mat-table
      class="msw-table mat-table"
      ng-reflect-data-source="[object Object],[object Object"
      role="grid"
    >
      
      <mat-header-row
        class="mat-header-row"
        role="row"
      >
        
        <mat-header-cell
          class="mat-header-cell cdk-column-name mat-column-name"
          role="columnheader"
        >
          נקודה
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-label mat-column-label"
          role="columnheader"
        >
          שם
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-icon mat-column-icon"
          role="columnheader"
        >
          סמל
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-type mat-column-type"
          role="columnheader"
        >
          סוג
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-delete mat-column-delete"
          role="columnheader"
        />
      </mat-header-row>
      
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          0
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           פטל כיתת 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          1
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           כיתת סחלב 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          2
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           כיתת קג׳בובו 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      
    </mat-table>
    <a
      class="add-point"
    >
      + הוסף איזור
    </a>
  </div>
</app-map-points-view>
`;

exports[`MapPointsViewComponent with class info should match snapshot 1`] = `
<app-map-points-view
  delete={[Function EventEmitter]}
  dialog={[Function MatDialog]}
  displayedColumns={[Function Array]}
  locations={[Function Array]}
  update={[Function EventEmitter]}
>
  <div
    class="msw-table-container"
  >
    <mat-table
      class="msw-table mat-table"
      ng-reflect-data-source="[object Object],[object Object"
      role="grid"
    >
      
      <mat-header-row
        class="mat-header-row"
        role="row"
      >
        
        <mat-header-cell
          class="mat-header-cell cdk-column-name mat-column-name"
          role="columnheader"
        >
          נקודה
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-label mat-column-label"
          role="columnheader"
        >
          שם
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-icon mat-column-icon"
          role="columnheader"
        >
          סמל
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-type mat-column-type"
          role="columnheader"
        >
          סוג
        </mat-header-cell>
        <mat-header-cell
          class="mat-header-cell cdk-column-delete mat-column-delete"
          role="columnheader"
        />
      </mat-header-row>
      
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          0
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           פטל כיתת 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          1
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           כיתת סחלב 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      <mat-row
        class="mat-row"
        role="row"
      >
        
        <mat-cell
          class="mat-cell cdk-column-name mat-column-name"
          role="gridcell"
        >
          2
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-label mat-column-label"
          role="gridcell"
        >
           כיתת קג׳בובו 
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-icon mat-column-icon"
          role="gridcell"
        >
          <img
            alt=""
            src="../../../assets/MSW Symbols/.png"
          />
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-type mat-column-type"
          role="gridcell"
        >
          
        </mat-cell>
        <mat-cell
          class="mat-cell cdk-column-delete mat-column-delete"
          role="gridcell"
        >
          <button
            class="delete-row-button mat-icon-button mat-primary"
            color="primary"
            data-test-id="delete-point-button"
            mat-icon-button=""
            ng-reflect-color="primary"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons"
              >
                 delete 
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-cell>
      </mat-row>
      
    </mat-table>
    <a
      class="add-point"
    >
      + הוסף איזור
    </a>
  </div>
</app-map-points-view>
`;
